src/Util/Temporal.php

```diff
@@ -0,0 +1,513 @@
+	 *
+	 * @todo Once browser support is better this could probably be replaced with
+	 * native HTML5 date picker.
+	 */
+	public static function getDateTimeField($min, $max, $default, $label, $id = 'datetimepicker', $pickdate = true,
+		$picktime = true, $minfrom = '', $maxfrom = '', $required = false)
+	{
+		// First day of the week (0 = Sunday)
+		$firstDay = PConfig::get(local_user(), 'system', 'first_day_of_week', 0);
+
+		$lang = substr(L10n::getBrowserLanguage(), 0, 2);
+
+		// Check if the detected language is supported by the picker
+		if (!in_array($lang,
+				["ar", "ro", "id", "bg", "fa", "ru", "uk", "en", "el", "de", "nl", "tr", "fr", "es", "th", "pl", "pt", "ch", "se", "kr",
+				"it", "da", "no", "ja", "vi", "sl", "cs", "hu"])) {
+			$lang = Config::get('system', 'language', 'en');
+		}
+
+		$o = '';
+		$dateformat = '';
+
+		if ($pickdate) {
+			$dateformat .= 'Y-m-d';
+		}
+
+		if ($pickdate && $picktime) {
+			$dateformat .= ' ';
+		}
+
+		if ($picktime) {
+			$dateformat .= 'H:i';
+		}
+
+		$minjs = $min ? ",minDate: new Date({$min->getTimestamp()}*1000), yearStart: " . $min->format('Y') : '';
+		$maxjs = $max ? ",maxDate: new Date({$max->getTimestamp()}*1000), yearEnd: " . $max->format('Y') : '';
+
+		$input_text = $default ? date($dateformat, $default->getTimestamp()) : '';
+		$defaultdatejs = $default ? ",defaultDate: new Date({$default->getTimestamp()}*1000)" : '';
+
+		$pickers = '';
+		if (!$pickdate) {
+			$pickers .= ', datepicker: false';
+		}
+
+		if (!$picktime) {
+			$pickers .= ',timepicker: false';
+		}
+
+		$extra_js = '';
+		$pickers .= ",dayOfWeekStart: " . $firstDay . ",lang:'" . $lang . "'";
+		if ($minfrom != '') {
+			$extra_js .= "\$('#id_$minfrom').data('xdsoft_datetimepicker').setOptions({onChangeDateTime: function (currentDateTime) { \$('#id_$id').data('xdsoft_datetimepicker').setOptions({minDate: currentDateTime})}})";
+		}
+
+		if ($maxfrom != '') {
+			$extra_js .= "\$('#id_$maxfrom').data('xdsoft_datetimepicker').setOptions({onChangeDateTime: function (currentDateTime) { \$('#id_$id').data('xdsoft_datetimepicker').setOptions({maxDate: currentDateTime})}})";
+		}
+
+		$readable_format = $dateformat;
+		$readable_format = str_replace('Y', 'yyyy', $readable_format);
+		$readable_format = str_replace('m', 'mm', $readable_format);
+		$readable_format = str_replace('d', 'dd', $readable_format);
+		$readable_format = str_replace('H', 'HH', $readable_format);
+		$readable_format = str_replace('i', 'MM', $readable_format);
+
+		$tpl = get_markup_template('field_input.tpl');
+		$o .= replace_macros($tpl,
+			[
+			'$field' => [
+				$id,
+				$label,
+				$input_text,
+				'',
+				$required ? '*' : '',
+				'placeholder="' . $readable_format . '"'
+			],
+		]);
+
+		$o .= "<script type='text/javascript'>";
+		$o .= "\$(function () {var picker = \$('#id_$id').datetimepicker({step:5,format:'$dateformat' $minjs $maxjs $pickers $defaultdatejs}); $extra_js})";
+		$o .= "</script>";
```

Comment:
This could probably be moved to the template as well.

Reply:
Not really, the `field_input.tpl` template is really generic, so a specific bit of Javascript about a datetimepicker wouldn't be welcome. Does it deserve its own template? Probably, but I'm not in a rush to add it.

Reply:
Ok, I have to create a specific template for this field then.

Reply:
Done.